jujusvg_test.go

```diff
@@ -54,19 +59,86 @@ func iconURL(ref *charm.Reference) string {
 	return "http://0.1.2.3/" + ref.Path() + ".svg"
 }
 
+type emptyFetcher struct{}
+
+func (f *emptyFetcher) FetchIcons(*charm.BundleData) (map[string]string, error) {
+	return map[string]string{}, nil
+}
+
+type errFetcher struct{}
```

Comment:
or even:

   type errFetcher string

...

    return nil, fmt.Errorf("%s", f)

which would allow the expected error to be mentioned more explicitly in the tests.